### PR TITLE
[Rollouts] Feature rollouts merge to main

### PR DESCRIPTION
**Pull request by themiswang**
Rollouts feature branch merge back to main.

lunchcal: launch/4286643

### CODE REVIEW

File: ClientApp/Podfile
Comment (paulb777) [inline ClientApp/Podfile:1]: First two files shouldn't be added. Maybe the result of a bad merge after the Integration Tests were moved to the IntegrationTesting directory?

File: Crashlytics/Crashlytics/Controllers/FIRCLSRolloutsPersistenceManager.h
Comment (paulb777) [inline Crashlytics/Crashlytics/Controllers/FIRCLSRolloutsPersistenceManager.h:19]: nit: "CocoaPods" and throughout

File: Crashlytics/Crashlytics/Handlers/FIRCLSException.mm
Comment (paulb777) [inline Crashlytics/Crashlytics/Handlers/FIRCLSException.mm:282]: nit: Also

File: Crashlytics/Crashlytics/Rollouts/CrashlyticsRemoteConfigManager.swift
Comment (paulb777) [inline Crashlytics/Crashlytics/Rollouts/CrashlyticsRemoteConfigManager.swift:52]: Writing

File: Crashlytics/Crashlytics/Rollouts/CrashlyticsRemoteConfigManager.swift
Comment (paulb777) [inline Crashlytics/Crashlytics/Rollouts/CrashlyticsRemoteConfigManager.swift:63]: inserted

File: Crashlytics/Crashlytics/Rollouts/CrashlyticsRemoteConfigManager.swift
Comment (paulb777) [inline Crashlytics/Crashlytics/Rollouts/CrashlyticsRemoteConfigManager.swift:71]: Update the 3 `debugPrint`s before launch

File: FirebaseRemoteConfigInterop.podspec
Comment (paulb777) [inline FirebaseRemoteConfigInterop.podspec:24]: Should be 1.12.0 now

File: ClientApp/Podfile
Comment (themiswang) [inline ClientApp/Podfile:1]: @paulb777 have we delete this file? Probably include here because of some rebase conflict I had before.

File: CocoapodsIntegrationTest/TestEnvironments/Cocoapods_multiprojects_frameworks/Podfile
Comment (themiswang) [inline CocoapodsIntegrationTest/TestEnvironments/Cocoapods_multiprojects_frameworks/Podfile:1]: Also this one @paulb777

**Comment by paulb777** [comment]
The new interop pod should be added to https://github.com/firebase/firebase-ios-sdk/blob/main/ReleaseTooling/Sources/FirebaseManifest/FirebaseManifest.swift

**Comment by themiswang** [comment]
> The new interop pod should be added to https://github.com/firebase/firebase-ios-sdk/blob/main/ReleaseTooling/Sources/FirebaseManifest/FirebaseManifest.swift

Thank your for the reference! I will ask for another review prior to the release for M145

**Comment by themiswang** [comment]
Hey @paulb777, our launchcal all bits are flipped. We are ready for merging back to main. There is 2 unit tests failed related to release tool and the related change is here: https://github.com/firebase/firebase-ios-sdk/pull/12464. Let me know if there is anything need to do on my side and we are ready to go!

File: FirebaseRemoteConfigSwift/Tests/SwiftAPI/FirebaseRemoteConfigSwift_APIBuildTests.swift
Comment (paulb777) [inline FirebaseRemoteConfigSwift/Tests/SwiftAPI/FirebaseRemoteConfigSwift_APIBuildTests.swift:26]: This test should remain since it's still part of the public API

File: Crashlytics/CHANGELOG.md
Comment (paulb777) [inline Crashlytics/CHANGELOG.md:3]: As a an app developer, I don't know what this means. What kind of information is reported?

File: FirebaseRemoteConfig/CHANGELOG.md
Comment (paulb777) [inline FirebaseRemoteConfig/CHANGELOG.md:2]: Shouldn't be developer visible, so can be removed.

File: FirebaseRemoteConfig/Sources/RCNConfigDBManager.h
Comment (paulb777) [inline FirebaseRemoteConfig/Sources/RCNConfigDBManager.h:117]: Reword

**Review by paulb777 — COMMENTED** [review]
Thanks for working through all the details on this @themiswang.

I have a few nits/questions and then LGTM

We can stage FirebaseRemoteConfigInterop after merge and that should address the zip builder ci failures.

File: Crashlytics/CHANGELOG.md
Comment (themiswang) [reply Crashlytics/CHANGELOG.md:3]: We had similar changelog on Android, cc @danasilver for details.

File: FirebaseRemoteConfigSwift/Tests/SwiftAPI/FirebaseRemoteConfigSwift_APIBuildTests.swift
Comment (themiswang) [reply FirebaseRemoteConfigSwift/Tests/SwiftAPI/FirebaseRemoteConfigSwift_APIBuildTests.swift:26]: cc @danasilver, @ddnan

File: FirebaseRemoteConfig/Sources/RCNConfigDBManager.h
Comment (themiswang) [reply FirebaseRemoteConfig/Sources/RCNConfigDBManager.h:117]: @danasilver, @ddnan have the fix here: https://github.com/firebase/firebase-ios-sdk/pull/1250, mind to take a look?

File: FirebaseRemoteConfig/CHANGELOG.md
Comment (danasilver) [reply FirebaseRemoteConfig/CHANGELOG.md:2]: Should we remove the whole entry for Remote Config?

File: FirebaseRemoteConfigSwift/Tests/SwiftAPI/FirebaseRemoteConfigSwift_APIBuildTests.swift
Comment (ddnan) [reply FirebaseRemoteConfigSwift/Tests/SwiftAPI/FirebaseRemoteConfigSwift_APIBuildTests.swift:26]: Working on a fix

File: Crashlytics/CHANGELOG.md
Comment (danasilver) [reply Crashlytics/CHANGELOG.md:3]: Yeah, this is the same changelog entry that we used for Android. The information is metadata about active Remote Config rollouts to aid with crash debugging.

Maybe we can say "Added support for Crashlytics to report metadata about Remote Config keys and values." since rollouts isn't available in the console yet and might be confusing to reference

**Review by danasilver — APPROVED** [review]

File: Crashlytics/CHANGELOG.md
Comment (themiswang) [reply Crashlytics/CHANGELOG.md:3]: Change: https://github.com/firebase/firebase-ios-sdk/pull/12515

File: FirebaseRemoteConfig/CHANGELOG.md
Comment (themiswang) [reply FirebaseRemoteConfig/CHANGELOG.md:2]: Discussed offline, we think it is still worth for developers to know which version we have a major update so we will keep the release note here

File: FirebaseRemoteConfigSwift/Tests/SwiftAPI/FirebaseRemoteConfigSwift_APIBuildTests.swift
Comment (ddnan) [reply FirebaseRemoteConfigSwift/Tests/SwiftAPI/FirebaseRemoteConfigSwift_APIBuildTests.swift]: Fixed in https://github.com/firebase/firebase-ios-sdk/pull/12503

**Review by samedson — APPROVED** [review]

**Pull request merged**
